src/common/components/repository-row/dropdowns/remove-repo-dropdown.js

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+
+import { Row, Data, Dropdown } from '../../vanilla/table-interactive';
+import Button from '../../vanilla/button';
+import { WarningIcon } from '../icons';
+
+import styles from './dropdowns.css';
+
+const RemoveRepoDropdown = (props) => {
+  const { latestBuild, registeredName } = props;
+  const { onCancelClick, onRemoveClick } = props;
+
+  let warningText;
+  if (latestBuild) {
+    warningText = (
+      'Removing this repo will delete all its builds and build logs.'
+    );
+  } else {
+    warningText = (
+      'Are you sure you want to remove this repo from the list?'
+    );
+  }
```

Comment:
Looking at it now it seems it could be moved up to the parent component and only passed as a string here.

This will remove the need for a bunch of store props passed into this component.